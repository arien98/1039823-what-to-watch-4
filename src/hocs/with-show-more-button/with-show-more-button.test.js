import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withButton from "./with-show-more-button.js";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withButton(MockComponent);

it(`with button renders correctly`, () => {
  const filmsData = [
    {
      id: 2,
      title: `Bohemian Rhapsody`,
      poster: `img/bohemian-rhapsody.jpg`,
    },
    {
      id: 3,
      title: `Macbeth`,
      poster: `img/macbeth.jpg`,
    },
    {
      id: 4,
      title: `Aviator`,
      poster: `img/aviator.jpg`,
    },
    {
      id: 5,
      title: `Bohemian Rhapsody`,
      poster: `img/bohemian-rhapsody.jpg`,
    },
    {
      id: 6,
      title: `Macbeth`,
      poster: `img/macbeth.jpg`,
    },
    {
      id: 7,
      title: `Aviator`,
      poster: `img/aviator.jpg`,
    },
    {
      id: 8,
      title: `Bohemian Rhapsody`,
      poster: `img/bohemian-rhapsody.jpg`,
    },
    {
      id: 9,
      title: `Macbeth`,
      poster: `img/macbeth.jpg`,
    },
    {
      id: 10,
      title: `Aviator`,
      poster: `img/aviator.jpg`,
    },
  ];

  const tree = renderer.create(
      <MockComponentWrapped
        filmsData={filmsData}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
