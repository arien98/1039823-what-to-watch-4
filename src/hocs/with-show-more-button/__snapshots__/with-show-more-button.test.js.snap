// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with button renders correctly 1`] = `
<div>
  
</div>
`;
